beacon-chain/p2p/watch_peers.go

```diff
@@ -5,15 +5,33 @@ import (
 
 	"github.com/libp2p/go-libp2p/core/host"
 	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/p2p/peers"
 )
 
 // ensurePeerConnections will attempt to reestablish connection to the peers
 // if there are currently no connections to that peer.
-func ensurePeerConnections(ctx context.Context, h host.Host, peers ...string) {
-	if len(peers) == 0 {
+func ensurePeerConnections(ctx context.Context, h host.Host, peers *peers.Status, relayNodes ...string) {
+	// every time reset peersToWatch, add RelayNodes and trust peers
+	var peersToWatch []string
+	trustedPeers := peers.GetTrustedPeers()
+	peersToWatch = append(peersToWatch, relayNodes...)
+	for _, trustedPeer := range trustedPeers {
+		address, err := peers.Address(trustedPeer)
+
+		// avoid invalid trusted peers
+		if err != nil || address == nil {
+			continue
+		}
+
+		// any more appropriate way ?
+		peer := address.String() + "/p2p/" + trustedPeer.String()
```

Comment:
Why is this being added in like this ?

Reply:
This seems incorrect, you need to construct the multiaddress so that we can then dial it in. You can look for examples in this package on how we do it.